### PR TITLE
add request and response counters for http versions

### PR DESCRIPTION
**Pull request by szuecs**
#735 add counters for http versions for upstream and downstream connections

```
        incoming       outgoing
[client] <---> [proxy] <---> [backend]

% curl http://localhost:9090/ --http1.0
<h1>Hello</h1>
% curl http://localhost:9090/ --http1.0
<h1>Hello</h1>
% curl http://localhost:9090/ --http1.1
<h1>Hello</h1>

% curl -s localhost:9911/metrics | jq .counters
{
  "skipper.outgoing.HTTP/1.1": {
    "count": 3
  },
  "skipper.incoming.HTTP/1.0": {
    "count": 2
  },
  "skipper.incoming.HTTP/1.1": {
    "count": 1
  }
}

```

What do you think, does it makes sense to measure these?

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -689,6 +689,7 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 	if req.Header.Get("Connection") == "Keep-Alive" {
 		req.Header.Set("Connection", "keep-alive")
 	}
+	p.metrics.IncCounter("downstream." + req.Proto)
```
Comment (aryszka) [inline proxy/proxy.go]: upstream and downstream may be ambiguous. Can we find a different key? One suggestion: request.incoming and request.outgoing?

Comment (szuecs) [reply proxy/proxy.go]: ok

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**